test/parallel/test-os-process-priority.js

```diff
@@ -0,0 +1,133 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const os = require('os');
+const {
+  PRIORITY_LOW,
+  PRIORITY_BELOW_NORMAL,
+  PRIORITY_NORMAL,
+  PRIORITY_ABOVE_NORMAL,
+  PRIORITY_HIGH,
+  PRIORITY_HIGHEST
+} = os.constants.priority;
+
+// Validate priority constants.
+assert.strictEqual(typeof PRIORITY_LOW, 'number');
+assert.strictEqual(typeof PRIORITY_BELOW_NORMAL, 'number');
+assert.strictEqual(typeof PRIORITY_NORMAL, 'number');
+assert.strictEqual(typeof PRIORITY_ABOVE_NORMAL, 'number');
+assert.strictEqual(typeof PRIORITY_HIGH, 'number');
+assert.strictEqual(typeof PRIORITY_HIGHEST, 'number');
+
+// Test pid type validation.
+[null, true, false, 'foo', {}, [], /x/].forEach((pid) => {
+  const errObj = {
+    code: 'ERR_INVALID_ARG_TYPE',
+    message: /The "pid" argument must be of type number\./
+  };
+
+  common.expectsError(() => {
+    os.setPriority(pid, PRIORITY_NORMAL);
+  }, errObj);
+
+  common.expectsError(() => {
+    os.getPriority(pid);
+  }, errObj);
+});
+
+// Test pid range validation.
+[NaN, Infinity, -Infinity, 3.14, 2 ** 32].forEach((pid) => {
+  const errObj = {
+    code: 'ERR_OUT_OF_RANGE',
+    message: /The value of "pid" is out of range\./
+  };
+
+  common.expectsError(() => {
+    os.setPriority(pid, PRIORITY_NORMAL);
+  }, errObj);
+
+  common.expectsError(() => {
+    os.getPriority(pid);
+  }, errObj);
+});
+
+// Test priority type validation.
+[null, true, false, 'foo', {}, [], /x/].forEach((priority) => {
+  common.expectsError(() => {
+    os.setPriority(0, priority);
+  }, {
+    code: 'ERR_INVALID_ARG_TYPE',
+    message: /The "priority" argument must be of type number\./
+  });
+});
+
+// Test priority range validation.
+[NaN, Infinity, -Infinity, 3.14, 2 ** 32].forEach((priority) => {
+  common.expectsError(() => {
+    os.setPriority(0, priority);
+  }, {
+    code: 'ERR_OUT_OF_RANGE',
+    message: /The value of "priority" is out of range\./
+  });
+});
+
+// Test priorities that are int32s outside of the "nice" range.
+[PRIORITY_HIGHEST - 1, PRIORITY_LOW + 1].forEach((priority) => {
+  common.expectsError(() => {
+    os.setPriority(0, priority);
+  }, {
+    code: 'ERR_SYSTEM_ERROR',
+    message: /A system error occurred: uv_os_setpriority returned EINVAL/
```

Comment:
I feel like it would be better for our users if we checked the range in JS. We could throw a more specific RangeError.

Reply:
Agreed. A specific `RangeError` with a friendlier error message would be good.